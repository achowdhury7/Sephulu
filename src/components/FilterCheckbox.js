import React, { Component } from 'react'

const FilterCheckbox = (props) => {
  return (
    <label className={props.labelClass}>
      <input name={props.name} type="checkbox" onChange={props.onChange} />
      {props.label}
    </label>
  )
}

export default FilterCheckbox