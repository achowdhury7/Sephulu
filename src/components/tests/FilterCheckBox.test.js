import React from 'react'
import { shallow, mount, render } from 'enzyme'

import FilterCheckbox from '../FilterCheckbox'

test('render a checkbox', () => {
  const wrapper = shallow(
    <FilterCheckbox />
  )
  expect(wrapper).toMatchSnapshot()
})

test('render a checkbox label', () => {
  const wrapper = mount(
    <FilterCheckbox label="testlabel" />
  )
  expect(wrapper.prop('label')).toEqual('testlabel')
})

test('pass selected value to onChange handler', () => {
  const value = {}
  const onChange = jest.fn()
  const wrapper = shallow(
    <FilterCheckbox onChange={onChange} />
  )

  expect(wrapper).toMatchSnapshot()

  wrapper.find('input').simulate('change', value)
  expect(onChange).toBeCalledWith(value)
})
