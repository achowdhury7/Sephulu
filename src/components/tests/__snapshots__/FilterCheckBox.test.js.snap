// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass selected value to onChange handler 1`] = `
<label>
  <input
    onChange={[MockFunction]}
    type="checkbox"
  />
</label>
`;

exports[`render a checkbox 1`] = `
<label>
  <input
    type="checkbox"
  />
</label>
`;
